db/judge/SolidityJudge.js

```diff
@@ -168,14 +167,14 @@ class SolidityJudge extends BaseJudge {
   promisifyTransaction (tx, args) {
     return new Promise((resolve, reject) => {
       tx({ ...args })
-      .on('confirmation', (confNumber, receipt) => {
-        if (confNumber === 1) {
-          resolve(receipt)
-        }
-      })
-      .on('error', (err) => {
-        reject(err)
-      })
+        .on('confirmation', (confNumber, receipt) => {
+          if (confNumber === 1) {
```

Comment:
this could be `>= 1` instead to be more defensive

Reply:
maybe provide a timeout as well?

Reply:
Ok! The confirmation event will fire 24 times. Promises are resolved once so it is ok. Why we need a timeout ?